### DISCUSSION TITLE
ActiveSelection ref in v6

### DISCUSSION THREAD

**Discussion by m3360202** [fabricjs/fabric.js#9066] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta5

### Node Version (if applicable)

None

### Link To Reproduction

record video downfloor

### Steps To Reproduce

1. use new activeselection() ,const target = new activeselection(canvas.getObjects(),{canvas});
2. setActiveSelection(target); canvas.requestRenderAll()
3.click this area or drag,the activeselection will be unseted

### Expected Behavior

can drag or click to select this activeselection,as mouse select behavior

### Actual Behavior

https://github.com/fabricjs/fabric.js/assets/36294167/9524ce8a-babd-481e-9d31-11d577a7f718

**Comment by m3360202** [comment]
![image](https://github.com/fabricjs/fabric.js/assets/36294167/dd443176-c1f3-4a2f-acb0-3ac2b55e7d26)
this is my code、

**Comment by m3360202** [comment]
I've fix this problem as the image downfloor,it seems that ,we need check if the this._activeSelection === this._activeObject
![QQ截图20230706223305](https://github.com/fabricjs/fabric.js/assets/36294167/f2cbc17e-3847-4dbe-bc4b-5de2829d3eda)
this problem seems has not been fix in v6 beta10 yet

**Comment by xorb** [comment]
I have the exact same issue here:

Creating a selection with selectionBox works fine, but, if you manually create a selection and drag it, something weird happens. 

https://codesandbox.io/p/sandbox/vigorous-cache-sf3mll?file=%2Fsrc%2FApp.tsx%3A73%2C16

**Reply by ShaMan123** [reply]
https://codesandbox.io/p/sandbox/fabric-artboard-forked-rxkvm4?file=%2Fsrc%2FApp.tsx%3A110%2C10

use canvas.getActiveSelection() => it is a constant ref so we can listen to events etc.

**Comment by xorb** [comment]
@ShaMan123 can you take a look ?

**Reply by m3360202** [reply]
Hi bro，u can fix fabric js lib src/canvas/selectableCanvas   ,rewrite the function getActiveObjects() as this:

```
getActiveObjects(): FabricObject[] {
    const active = this._activeObject;
    if (active) {
      if (active === this._activeSelection) {
        return [...(active as ActiveSelection)._objects];
      } else {
        if (active.type === 'activeselection' && active._objects && active._objects.length>0) {
          return [...(active as ActiveSelection)._objects];
        } else {
          return [active];
        }

      }
    }
    return [];
  }
```

**Reply by xorb** [reply]
Yeah, that seems to be working fine, thanks. 
```ts
  getActiveObjects(): FabricObject[] {
    const active = this._activeObject;
    if (active) {
      if (
        active === this._activeSelection ||
        active instanceof ActiveSelection
      ) {
        return [...(active as ActiveSelection)._objects];
      } else {
        return [active];
      }
    }
    return [];
  }
```

**Reply by ShaMan123** [reply]
bad workaround

**Reply by ShaMan123** [reply, marked as answer]
https://github.com/fabricjs/fabric.js/discussions/9066#discussioncomment-6391446
